### PR TITLE
bugfix(build): Fixes the case where the addon has been included more than once

### LINKED ISSUE #173: SyntaxError: ember-frost-core/components/frost-button.js: Unexpected token (104:2)
### Problem Description
When using `ember-decorators` version 1.3.0 or 1.3.1 (only versions I have tested this in) in an add-on `ember-frost-core` (Ember CLI version 2.16.2, ember-cli-babel version 6.8.2), I am still finding it necessary to set `transform-decorators-legacy` and `transform-class-properties` in the `init()` of the addon's `index.js`.

https://github.com/juwara0/ember-frost-core/blob/emberDecorators_1-3-1/index.js#L119-L125

If I do not then I get the following error in the terminal:
```
->  ember serve
Could not start watchman
Visit https://ember-cli.com/user-guide/#watchman for more info.
Livereload server on http://localhost:7020
The Broccoli Plugin: [BroccoliMergeTrees: Addon#compileAddon(ember-frost-core) ] failed with:
SyntaxError: ember-frost-core/components/frost-button.js: Unexpected token (104:2)
  102 |   // == Computed Properties ===================================================
  103 | 
> 104 |   @readOnly
      |   ^
  105 |   @computed('icon', 'text')
  106 |   /**
  107 |    * Determine whether or not button is text only (no icon)
    at Parser.pp$5.raise (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$1.parseDecorator (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:1939:10)
    at Parser.pp$3.parseObj (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3983:28)
    at Parser.pp$3.parseExprAtom (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3719:19)
    at Parser.pp$3.parseExprSubscripts (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli-babel/node_modules/babylon/lib/index.js:3344:19)

The broccoli plugin was instantiated at: 
    at BroccoliMergeTrees.Plugin (/Users/Micarrol/Devel/ember-frost-core/node_modules/broccoli-plugin/index.js:7:31)
    at new BroccoliMergeTrees (/Users/Micarrol/Devel/ember-frost-core/node_modules/broccoli-merge-trees/index.js:16:10)
    at Function.BroccoliMergeTrees [as _upstreamMergeTrees] (/Users/Micarrol/Devel/ember-frost-core/node_modules/broccoli-merge-trees/index.js:10:53)
    at mergeTrees (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/lib/broccoli/merge-trees.js:85:33)
    at Class.compileAddon (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/lib/models/addon.js:1092:12)
    at Class.treeForAddon (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/lib/models/addon.js:746:26)
    at Class.treeForAddon (/Users/Micarrol/Devel/ember-frost-core/index.js:148:48)
    at Class.superWrapper [as treeForAddon] (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/node_modules/core-object/lib/assign-properties.js:34:20)
    at Class._treeFor (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/lib/models/addon.js:557:33)
    at Class.treeFor (/Users/Micarrol/Devel/ember-frost-core/node_modules/ember-cli/lib/models/addon.js:517:21)
```

`npm ls ember-cli-babel`
```
->  npm ls ember-cli-babel
ember-frost-core@2.0.0 /Users/Micarrol/Devel/ember-frost-core
├── ember-cli-babel@6.8.2 
├─┬ ember-cli-chai@0.4.3
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-cli-code-coverage@0.3.12
│ └── ember-cli-babel@5.2.4 
├─┬ ember-cli-mirage@0.2.5
│ ├── ember-cli-babel@5.2.4 
│ ├─┬ ember-inflector@1.12.1
│ │ └── ember-cli-babel@5.2.4 
│ └─┬ ember-lodash@0.0.9
│   └── ember-cli-babel@5.2.4 
├─┬ ember-cli-mocha@0.14.4
│ ├── ember-cli-babel@6.8.2  deduped
│ └─┬ ember-cli-test-loader@2.2.0
│   └── ember-cli-babel@6.8.2  deduped
├─┬ ember-cli-notifications@4.2.1
│ ├── ember-cli-babel@5.2.4 
│ ├─┬ ember-css-modules@0.7.2
│ │ ├── ember-cli-babel@6.8.2  deduped
│ │ └─┬ ember-getowner-polyfill@1.2.5
│ │   └── ember-cli-babel@5.2.4 
│ └─┬ ember-runtime-enumerable-includes-polyfill@2.0.0
│   └── ember-cli-babel@6.8.2  deduped
├─┬ ember-cli-shims@1.1.0
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-concurrency@0.8.10
│ ├── ember-cli-babel@6.8.2  deduped
│ └─┬ ember-maybe-import-regenerator@0.1.6
│   └── ember-cli-babel@6.8.2  deduped
├─┬ ember-data@2.16.3
│ ├── ember-cli-babel@6.8.2  deduped
│ └─┬ ember-inflector@2.0.1
│   └── ember-cli-babel@6.8.2  deduped
├─┬ ember-decorators@1.3.1
│ ├── ember-cli-babel@6.8.2  deduped
│ └─┬ ember-macro-helpers@0.16.3
│   ├── ember-cli-babel@6.8.2  deduped
│   └─┬ ember-weakmap@2.0.0
│     └── ember-cli-babel@5.2.4 
├─┬ ember-elsewhere@1.0.2
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-export-application-global@2.0.0
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-frost-test@2.1.2
│ └── ember-cli-babel@5.2.4 
├─┬ ember-hook@1.4.2
│ ├── ember-cli-babel@6.8.2  deduped
│ └─┬ ember-get-config@0.2.3
│   └── ember-cli-babel@6.8.2  deduped
├─┬ ember-load-initializers@1.0.0
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-prop-types@4.0.1
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-resolver@4.5.0
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-sinon@0.7.0
│ └── ember-cli-babel@5.2.4 
├─┬ ember-spread@2.0.1
│ └── ember-cli-babel@6.8.2  deduped
├─┬ ember-test-utils@6.0.0
│ └── ember-cli-babel@5.2.4 
├─┬ ember-truth-helpers@2.0.0
│ └── ember-cli-babel@6.8.2  deduped
└─┬ markdown-code-highlighting@0.1.0
  └── ember-cli-babel@5.2.4 
```

I also noticed that I needed to have `ember-decorators` run before this addon: https://github.com/juwara0/ember-frost-core/blob/emberDecorators_1-3-1/package.json#L80-L82

### Expected Outcomes
I am not sure if this is a bug or if the above settings are correct and the README needs to capture these steps.

### PR DESCRIPTION
**Pull request by pzuraq**
Fixes #173 

The included hook only runs once per instance of the addon in the dependency graph. The babel transforms we are adding, however, are meant to be added to each addon that includes this addon as a dependency, as well as the root app. Because each addon has it's own version of ember-cli-babel, we have to make sure it happens for each of them.

To do this, we use the `init` hook instead of the `included` hook, since it runs each time the addon is created. Unfortunately, the init hook does not have any way to access the root app - that is only provided in the included hook. So, we still need to use included for the case where this addon is being used in an actual app.

TODO: Tests. We've had enough edge-cases and failures caused by misunderstandings of Ember-CLI, I think we need to figure out some way to test various addon inclusion scenarios to prevent issues like this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "babel-plugin-transform-decorators-legacy": "^1.3.4",
     "ember-cli-babel": "^6.0.0",
     "ember-cli-version-checker": "^2.0.0",
-    "ember-compatibility-helpers": "^0.1.0",
+    "ember-compatibility-helpers": "^0.1.2",
```
Comment (pzuraq) [inline package.json]: `ember-compatibility-helpers` was also affected by a similar bug, bumped the version explicitly just in case.

**Comment by rwjblue** [comment]
> The included hook only runs once per instance of the addon in the dependency graph.

This does not seem correct to me. We do not dedupe the included invocations in ember-cli...

**Comment by pzuraq** [comment]
After more thorough testing, I can definitely reproduce the issue in `ember-frost-core`, the library that this was reported in, but I can't reproduce this in any other setup of addons/apps 😐 I'm not sure what's going on with their setup that could cause addons to initialize but not be included, but yeah definitely appears that I was incorrect and this is not our issue

Still definitely want to setup some tests in the future to make sure we have all of this behavior formalized, I think that would reduce confusion as well. Will start looking into [ember-cli-addon-tests](https://github.com/tomdale/ember-cli-addon-tests) when I get some time.

Closing for now, will continue discussion on #173.
